### PR TITLE
[RPA 931.2] Excel SharePoint actions refactor

### PR DESCRIPTION
**Pull request by MRSIMM0**
## Description
Partial refactor of SharePoint excel actions - (only one drive)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

---

I agree to the terms of the RunBotics Contributor License Agreement.

### CODE REVIEW

File: runbotics/runbotics-common/src/actions/actions.ts [hunk 1]
```diff
@@ -1,3 +1,6 @@
+export const OneDrive = 'OneDrive';
+export const Sharepoint = 'Sharepoint';
```
Comment (preludium) [inline runbotics/runbotics-common/src/actions/actions.ts]: camelCase

File: runbotics/runbotics-desktop/src/action/microsoft/excel/excel.service.ts
Comment (preludium) [inline runbotics/runbotics-desktop/src/action/microsoft/excel/excel.service.ts:1]: spaces between functions plx

File: runbotics/runbotics-desktop/src/action/microsoft/excel/excel.service.ts [hunk 2]
```diff
@@ -2,22 +2,155 @@ import { Injectable } from '@nestjs/common';
...
+                break;
+            case OneDrive:
+                request = `/me/drive/root:/${this.session.sessionIdentifier}:/workbook`.concat(request);
+                break;
+            default:
+                throw new Error('Invalid platform');
+        }
+
+        return request;
+    }
```
Comment (preludium) [inline runbotics/runbotics-desktop/src/action/microsoft/excel/excel.service.ts]: do not mutate your function parameters, here you can just return inside cases

File: runbotics/runbotics-desktop/src/action/microsoft/excel/excel.service.ts [hunk 3]
```diff
@@ -2,22 +2,155 @@ import { Injectable } from '@nestjs/common';
...
+
+    private getColumnNumber(column: string): number {
+        return (column.length - 1) * 26 + (column.charCodeAt(column.length - 1) - 64);
```
Comment (preludium) [inline runbotics/runbotics-desktop/src/action/microsoft/excel/excel.service.ts:156]: magic numbers to me

File: runbotics/runbotics-desktop/src/action/microsoft/excel/excel.service.ts [hunk 4]
```diff
@@ -2,22 +2,155 @@ import { Injectable } from '@nestjs/common';
...
+    }
+
+    private async getDriveIdBySite(siteId: string) {
+        const url = `/sites/${siteId}/drives/?$select=id,name`;
+
+        const data = (await this.microsoftGraphService.get(url)) as any;
```
Comment (preludium) [inline runbotics/runbotics-desktop/src/action/microsoft/excel/excel.service.ts]: `as any` we don't do this here

File: runbotics/runbotics-desktop/src/action/microsoft/excel/excel.service.ts [hunk 5]
```diff
@@ -2,22 +2,155 @@ import { Injectable } from '@nestjs/common';
...
+
+        if (!Array.isArray(values)) {
+            values = JSON.parse(values) as any[][];
+        }
+
+        const newRange: WorkbookRangeUpdateBody = {
+            values,
+        };
+
+        return await this.microsoftGraphService.patch(this.createWorkbookUrl(request), newRange, {
```
Comment (preludium) [inline runbotics/runbotics-desktop/src/action/microsoft/excel/excel.service.ts]: no need to do `return await` just return a promise, thats it. Fix all similar occurrences

File: runbotics/runbotics-desktop/src/action/microsoft/excel/excel.types.ts [hunk 6]
```diff
@@ -0,0 +1,141 @@
+export type SessionIdentifier = number | string;
+export type WorksheetIdentifier = number | string;
+export type Platform = 'Sharepoint' | 'OneDrive';
```
Comment (preludium) [inline runbotics/runbotics-desktop/src/action/microsoft/excel/excel.types.ts]: it's `SharePoint`

File: runbotics/runbotics-desktop/src/action/microsoft/excel/excel.types.ts [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+    sessionIdentifier: SessionIdentifier;
+    worksheetIdentifier: WorksheetIdentifier;
+    workbookSessionInfo: WorkbookSessionInfo;
+    siteId?: string;
+    driveId?: string;
+}
+
+export interface WorkbookSessionInfo {
+    id: string;
+    persistChanges: boolean;
```
Comment (preludium) [inline runbotics/runbotics-desktop/src/action/microsoft/excel/excel.types.ts:16]: what does it do?

File: runbotics/runbotics-desktop/src/action/microsoft/excel/excel.types.ts
Comment (preludium) [inline runbotics/runbotics-desktop/src/action/microsoft/excel/excel.types.ts:1]: add spaces between interfaces

File: runbotics/runbotics-desktop/src/action/microsoft/excel/excel.types.ts [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+    quota?: object;
+    root?: object;
+    sharepointIds?: object;
+    special?: object;
+    system?: object;
+    webUrl?: string;
+}
+//https://learn.microsoft.com/en-us/graph/api/resources/driveitem?view=graph-rest-1.0
+export interface DriveItem {
+    audio?: object;
```
Comment (preludium) [inline runbotics/runbotics-desktop/src/action/microsoft/excel/excel.types.ts]: it should be `unknown` instead of `object` everywhere in this file

Comment (preludium) [reply runbotics/runbotics-desktop/src/action/microsoft/excel/excel.types.ts]: but if we use it then investigate what's inside

File: runbotics/runbotics-desktop/src/action/sharepoint/excel/sharepoint-excel.action-handler.ts [hunk 9]
```diff
@@ -1,76 +1,63 @@
...
+            input.persistChanges,
+            input.siteRelativePath,
+        );
     }
 
     async closeSession(
         input: SharepointTypes.SharepointExcelCloseSessionActionInput
     ): Promise<SharepointTypes.SharepointExcelCloseSessionActionOutput> {
-        return await this.microsoftService.closeSession();
+        return await this.excelService.closeSession();
```
Comment (preludium) [inline runbotics/runbotics-desktop/src/action/sharepoint/excel/sharepoint-excel.action-handler.ts]: no need to do `return await`

File: runbotics/runbotics-desktop/src/action/sharepoint/excel/types.ts [hunk 10]
```diff
@@ -32,7 +27,7 @@ export type SharepointExcelSetCellActionOutput = any;
 
 export type SharepointExcelUpdateRangeActionInput = {
     range: string;
-    values: any[][];
+    values: string | any[][];
```
Comment (preludium) [inline runbotics/runbotics-desktop/src/action/sharepoint/excel/types.ts:30]: but is it `any`? isn't it `(string | number)[][]`?

File: runbotics/runbotics-desktop/src/action/sharepoint/excel/types.ts [hunk 11]
```diff
@@ -3,25 +3,20 @@ import { DesktopRunRequest } from 'runbotics-sdk';
...
+| DesktopRunRequest<'sharepointExcel.openFile', SharepointExcelOpenFileActionInput>;
 
-
-export type SharepointOpenFromRootActionInput = {
-    filePath: string;
-    worksheetName: string;
+export interface SharepointExcelOpenFileActionInput {
+    platform:string;
+    filePath:string;
+    worksheetName:string;
```
Comment (preludium) [inline runbotics/runbotics-desktop/src/action/sharepoint/excel/types.ts]: space after `:`

Comment (preludium) [reply runbotics/runbotics-desktop/src/action/sharepoint/excel/types.ts]: try to pay attention to it

File: runbotics/runbotics-desktop/src/action/sharepoint/excel/types.ts [hunk 12]
```diff
@@ -3,25 +3,20 @@ import { DesktopRunRequest } from 'runbotics-sdk';
...
-| DesktopRunRequest<'sharepointExcel.getRange', SharepointExcelGetRangeActionInput>;
+| DesktopRunRequest<'sharepointExcel.getRange', SharepointExcelGetRangeActionInput>
+| DesktopRunRequest<'sharepointExcel.openFile', SharepointExcelOpenFileActionInput>;
 
-
-export type SharepointOpenFromRootActionInput = {
-    filePath: string;
-    worksheetName: string;
+export interface SharepointExcelOpenFileActionInput {
+    platform:string;
```
Comment (preludium) [inline runbotics/runbotics-desktop/src/action/sharepoint/excel/types.ts]: why `platform: string` if you know what we expect `SharePoint | OneDrive` a then you don't need to do `@ts-ignore`

**Review by preludium — CHANGES_REQUESTED** [review]

File: runbotics/runbotics-desktop/src/action/microsoft/excel/excel.types.ts (hunk 7 above)
Comment (MRSIMM0) [reply runbotics/runbotics-desktop/src/action/microsoft/excel/excel.types.ts:16]: This is response from /createSession 
id - sessionId used to improve performance
persistChanges - return if changes that you made is persisted

File: runbotics/runbotics-desktop/src/action/sharepoint/excel/types.ts (hunk 10 above)
Comment (MRSIMM0) [reply runbotics/runbotics-desktop/src/action/sharepoint/excel/types.ts:30]: It's string or any[][]. String is used when user input passed as array not as variable. And it should't be just number maybe number | string.

**Pull request closed without merging**